packages/block-library/src/query/edit.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Internal dependencies
+ */
+import QueryPanel from './query-panel';
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { debounce, isUndefined, pickBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockList,
+	BlockEditorProvider,
+	InspectorControls,
+	WritingFlow,
+} from '@wordpress/block-editor';
+import { cloneBlock, createBlock } from '@wordpress/blocks';
+import { PanelBody, Placeholder, Spinner } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { EntityProvider } from '@wordpress/core-data';
+import { withSelect } from '@wordpress/data';
+
+const defaultFields = [
+	'core/post-title',
+	'core/post-date',
+	'core/post-author',
+	'core/post-excerpt',
+];
+
+class Edit extends Component {
```

Comment:
Not very important but in new components we're trying to avoid Component in favor of hooks.

Reply:
I took a stab at that a while ago and ran into some trouble. Once I tackle the rest of the things you've highlighted it'll probably be easier to try again.